### PR TITLE
Memory game activity

### PR DESCRIPTION
**Pull request by Raxerz**

### CODE REVIEW

File: src/app/meds-n-labels/activity-2/activity-2.component.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+            if (this._clicks === 0) {
+                this._firstchoice = card;
+                document.images[card].src = this._baseImgPath + this._faces[card];
+                this._clicks = 1;
+            } else if (this._firstchoice !== card) {
+                this._clicks = 2;
+                this._secondchoice = card;
+                document.images[card].src = this._baseImgPath + this._faces[card];
+                this._obs = Observable.interval(500)
+                        .do(i => this.check() );
```
Comment (yukie1412) [inline src/app/meds-n-labels/activity-2/activity-2.component.ts]: spacing

File: src/app/meds-n-labels/activity-2/activity-2.component.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+            this._matches++;
+            if (this._matches === 8) {
+                this.successAlert();
+                this._status = {stage: 3, activity: 2};
+                this._dashboardService.updateProgressStatus(this._status).subscribe(response => {});
+            }
+        } else {
+            document.images[this._firstchoice].src = this.backcard;
+            document.images[this._secondchoice].src = this.backcard;
+            return;
```
Comment (yukie1412) [inline src/app/meds-n-labels/activity-2/activity-2.component.ts]: necessary?

File: src/app/meds-n-labels/meds-n-labels.component.scss [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+img {
+    height: 30%;
+    width: 100%;
+    cursor: pointer;
+}
+
+.image-table {
+    border: 0px solid rgba(0, 0, 0, 0);
```
Comment (yukie1412) [inline src/app/meds-n-labels/meds-n-labels.component.scss]: `border: 0;`

File: src/app/meds-n-labels/meds-n-labels.component.scss [hunk 4]
```diff
@@ -0,0 +1,26 @@
+h3{
```
Comment (yukie1412) [inline src/app/meds-n-labels/meds-n-labels.component.scss]: not sass standard

File: src/app/meds-n-labels/activity-2/activity-2.component.ts (hunk 2 above)
Comment (Raxerz) [inline src/app/meds-n-labels/activity-2/activity-2.component.ts]: Hmm yes, when 2 cards are open they have to revert back to the closed state and hence the 2 lines for each of those cards.

File: src/app/meds-n-labels/activity-2/activity-2.component.ts [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+     */
+    successAlert() {
+        swal(
+            'Good job!',
+            'You completed this activity!',
+            'success'
+        );
+    }
+
+    checkProgress() {
```
Comment (yukie1412) [inline src/app/meds-n-labels/activity-2/activity-2.component.ts]: this is just duplicated code with https://github.com/systers/PC-Prep-Kit/pull/25/files#diff-78168c46d826955cc8506302e6bba148R75

i'd create a global function `checkProgress(stage, activity)` instead

File: src/app/meds-n-labels/meds-n-labels.component.scss [hunk 6]
```diff
@@ -0,0 +1,48 @@
+@import "../styles/components/_variables";
+
+.stage-header{
```
Comment (yukie1412) [inline src/app/meds-n-labels/meds-n-labels.component.scss]: spacing

File: src/app/meds-n-labels/meds-n-labels.component.scss [hunk 7]
```diff
@@ -0,0 +1,48 @@
+@import "../styles/components/_variables";
```
Comment (yukie1412) [inline src/app/meds-n-labels/meds-n-labels.component.scss]: are you following the scss convention in this?

Comment (Raxerz) [inline src/app/meds-n-labels/meds-n-labels.component.scss]: This file was created by Shreyans, so I am not aware about that.

File: src/app/app.routing.ts [hunk 8]
```diff
@@ -14,6 +14,10 @@ import { NavbarComponent } from './navbar/navbar.component';
 import { ActivityintroComponent } from './activityintro/activityintro.component';
 import { MenuComponent } from './menu/menu.component';
 import { DragdropComponent} from './dragdrop/dragdrop.component';
+import { MedsNLabelsComponent } from './meds-n-labels/meds-n-labels.component';
+import { MemoryGameComponent } from './meds-n-labels/activity-2/activity-2.component';
+
+>>>>>>> Memory game activity
```
Comment (yukie1412) [inline src/app/app.routing.ts]: fix the conflict

File: src/app/meds-n-labels/activity-2/activity-2.component.ts [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+    }
+
+    /**
+     * Flip card on click
+     * @param {Number} card The index of the clicked element
+     */
+    choose(card) {
+            if (this._clicks === 2) {
+                return;
+            } else {
```
Comment (yukie1412) [inline src/app/meds-n-labels/activity-2/activity-2.component.ts]: unnecessary `else`

File: src/app/meds-n-labels/activity-2/activity-2.component.ts [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+            if (this._matches === 8) {
+                this._sharedData.customAlert('Good job!', 'You completed this activity!', 'success');
+                this._status = {stage: 3, activity: 2};
+                this._dashboardService.updateProgressStatus(this._status).subscribe(response => {});
+                this.activityComplete = true;
+            }
+        } else {
+            this.getImgElement(this._firstchoice).src = this.backcard;
+            this.getImgElement(this._secondchoice).src = this.backcard;
+            return;
```
Comment (yukie1412) [inline src/app/meds-n-labels/activity-2/activity-2.component.ts]: i'm still trying to understand the need of this `return`, this is the last line of the function, so it automatically returns/terminates. 
also, unnecessary `else`

File: src/app/meds-n-labels/meds-n-labels.component.scss (hunk 7 above)
Comment (yukie1412) [inline src/app/meds-n-labels/meds-n-labels.component.scss]: isn't this your pr?

Comment (Raxerz) [inline src/app/meds-n-labels/meds-n-labels.component.scss]: No @yukie1412  this was not my PR.

Comment (yukie1412) [inline src/app/meds-n-labels/meds-n-labels.component.scss]: then why are you including this file here?

**Comment by yukie1412** [comment]
i'm going to merge this PR for now.
but please coordinate between yourselves regard `src/app/meds-n-labels/meds-n-labels.component.scss`

**Pull request merged**